### PR TITLE
Read bios and local storage attributes

### PR DESCRIPTION
**Pull request by nabhajit-ray**
### Description
Read bios and local storage attributes 

### Issues Resolved
[#233 ]

### Check List
- [ ] New functionality includes testing.
  - [x]  All tests pass for go 1.11 + gofmt checks.
- [ ] New functionality has been documented in the README if applicable.
  - [ ] New functionality has been thoroughly documented in the examples (please include helpful comments).
  - [ ] New endpoints supported are updated in the endpoints-support.md file.
- [ ] Changes are documented in the CHANGELOG.

### CODE REVIEW

File: oneview/data_source_server_profile_template.go [hunk 1]
```diff
@@ -39,6 +39,58 @@ func dataSourceServerProfileTemplate() *schema.Resource {
...
+			"bios_option": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"manage_bios": {
+							Type:     schema.TypeBool,
+							Required: true,
+						},
+						// "compliance_control":{
```
Comment (AsisBagga) [inline oneview/data_source_server_profile_template.go:71]: please remove the commented compliance control if not required.

File: oneview/resource_server_profile.go [hunk 2]
```diff
@@ -1036,6 +1203,10 @@ func resourceServerProfileCreate(d *schema.ResourceData, meta interface{}) error
 
 		}
 	}
+
+	file, _ := json.MarshalIndent(serverProfile, "", " ")
```
Comment (AsisBagga) [inline oneview/resource_server_profile.go:1207]: please remove this

File: oneview/resource_server_profile.go [hunk 3]
```diff
@@ -18,6 +18,7 @@ import (
 	"github.com/HewlettPackard/oneview-golang/ov"
 	"github.com/HewlettPackard/oneview-golang/utils"
 	"github.com/hashicorp/terraform/helper/schema"
+	"io/ioutil"
```
Comment (AsisBagga) [inline oneview/resource_server_profile.go:21]: if not required, please remove this

File: oneview/resource_server_profile.go [hunk 4]
```diff
@@ -1036,6 +1203,10 @@ func resourceServerProfileCreate(d *schema.ResourceData, meta interface{}) error
 
 		}
 	}
+
+	file, _ := json.MarshalIndent(serverProfile, "", " ")
+	_ = ioutil.WriteFile("sp.json", file, 0644)
```
Comment (AsisBagga) [inline oneview/resource_server_profile.go:1208]: Same as above.

File: oneview/resource_server_profile.go [hunk 5]
```diff
@@ -1047,8 +1218,8 @@ func resourceServerProfileCreate(d *schema.ResourceData, meta interface{}) error
 			return err
 		}
 	}
-
-	return resourceServerProfileRead(d, meta)
+	return nil
+	//return resourceServerProfileRead(d, meta)
```
Comment (AsisBagga) [inline oneview/resource_server_profile.go:1222]: remove the commented line if not required.

File: oneview/resource_server_profile_template.go [hunk 6]
```diff
@@ -68,9 +68,13 @@ func resourceServerProfileTemplate() *schema.Resource {
 							Type:     schema.TypeBool,
 							Required: true,
 						},
+						// "compliance_control":{
```
Comment (AsisBagga) [inline oneview/resource_server_profile_template.go:71]: remove this if not required.

File: oneview/data_source_server_profile_template.go [hunk 7]
```diff
@@ -170,5 +371,83 @@ func dataSourceServerProfileTemplateRead(d *schema.ResourceData, meta interface{
...
+			"manage_bios":         spt.Bios.ManageBios,
+			"overridden_settings": overriddenSettings,
+		})
+
+		d.Set("bios_option", biosOptions)
+	}
+
+	localStorages := make([]map[string]interface{}, 0, 1)
+	localStorages = append(localStorages, map[string]interface{}{
+		//"manage_local_storage": serverProfile.LocalStorage.ManageLocalStorage,
```
Comment (AsisBagga) [inline oneview/data_source_server_profile_template.go:393]: any specific reason to why we are not capturing the manage_local_storage and initialize value, is not only required at time of creating the resource?

Comment (nabhajit-ray) [reply oneview/data_source_server_profile_template.go:393]: manage local storage and initialize value is not part of the local storage.
http://apidocs.vse.rdlabs.hpecorp.net/fusion/2200/synergy/public/#rest/server-profiles

Comment (nabhajit-ray) [reply oneview/data_source_server_profile_template.go:393]: > any specific reason to why we are not capturing the manage_local_storage and initialize value, is not only required at time of creating the resource?
Those fields are not part of local storage
http://apidocs.vse.rdlabs.hpecorp.net/fusion/2200/synergy/public/#rest/server-profiles

**Pull request closed without merging**